src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -736,6 +778,49 @@ private bool ArePointerTypesEqual(PointerTypeSymbol type, PointerTypeSymbol othe
                 return AreTypesEqual(type.PointedAtType, other.PointedAtType);
             }
 
+            private bool AreFunctionPointerTypesEqual(FunctionPointerTypeSymbol type, FunctionPointerTypeSymbol other)
+            {
+                var sig = type.Signature;
+                var otherSig = other.Signature;
+
+                ValidateFunctionPointerParamOrReturn(sig.ReturnTypeWithAnnotations, sig.RefKind, sig.RefCustomModifiers, allowOut: false);
+                ValidateFunctionPointerParamOrReturn(otherSig.ReturnTypeWithAnnotations, otherSig.RefKind, otherSig.RefCustomModifiers, allowOut: false);
+                if (sig.RefKind != otherSig.RefKind || !AreTypesEqual(sig.ReturnTypeWithAnnotations, otherSig.ReturnTypeWithAnnotations))
+                {
+                    return false;
+                }
+
+                return sig.Parameters.SequenceEqual(otherSig.Parameters, AreFunctionPointerParametersEqual);
+            }
+
+            private bool AreFunctionPointerParametersEqual(ParameterSymbol param, ParameterSymbol otherParam)
+            {
+                ValidateFunctionPointerParamOrReturn(param.TypeWithAnnotations, param.RefKind, param.RefCustomModifiers, allowOut: true);
+                ValidateFunctionPointerParamOrReturn(otherParam.TypeWithAnnotations, otherParam.RefKind, otherParam.RefCustomModifiers, allowOut: true);
+
+                return param.RefKind == otherParam.RefKind && AreTypesEqual(param.TypeWithAnnotations, otherParam.TypeWithAnnotations);
+            }
+
+            [Conditional("DEBUG")]
+            private static void ValidateFunctionPointerParamOrReturn(TypeWithAnnotations type, RefKind refKind, ImmutableArray<CustomModifier> refCustomModifiers, bool allowOut)
+            {
+                Debug.Assert(type.CustomModifiers.IsEmpty);
+                Debug.Assert(verifyRefModifiers(refCustomModifiers, refKind, allowOut));
+
+                static bool verifyRefModifiers(ImmutableArray<CustomModifier> modifiers, RefKind refKind, bool allowOut)
+                {
+                    Debug.Assert(RefKind.RefReadOnly == RefKind.In);
+                    switch (refKind)
+                    {
+                        case RefKind.RefReadOnly:
+                        case RefKind.Out when allowOut:
+                            return modifiers.Length == 1;
+                        default:
+                            return modifiers.IsDefaultOrEmpty;
```

Comment:
>IsDefaultOrEmpty [](start = 45, length = 16)

Can this be default?

Reply:
No.

---
In reply to: [426075093](https://github.com/dotnet/roslyn/pull/44256#discussion_r426075093) [](ancestors = 426075093)